python/tk_desktop/desktop_engine_site_implementation.py

```diff
@@ -213,6 +214,15 @@ def _on_proxy_closing(self):
         # Clear the UI, we can't launch anything anymore!
         self.desktop_window.clear_app_uis()
 
+    def _on_proxy_created(self):
+        """
+        Invoked when background process has created proxy
+        """
+        logger.debug("_on_proxy_created")
+        # Clears the project menu to prevent xtra engine init from  multi-populating
```

Comment:
How about `# Clears the project menu so the previous engine's actions are removed before adding new one`